OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -485,7 +485,7 @@ void UpdateCenterLocation(Actor self, Mobile mobile)
 
 				if (moveFraction >= MoveFractionTotal)
 					mobile.Facing = ToFacing;
-				else
+				else if (FromFacing != ToFacing)
```

Comment:
I suspect that this is putting a bandaid over some other bug, which I expect will be breaking something that we don't realize yet!

If this were just a rounding issue then I would expect that the following would behave the same as the current PR:
```csharp
if (moveFraction >= MoveFractionTotal)
	mobile.Facing = ToFacing;
else if (FromFacing == ToFacing)
	mobile.Facing = ToFacing;
else
	mobile.Facing = WAngle.Lerp(FromFacing, ToFacing, moveFraction, MoveFractionTotal);
```
However, it actually keeps the behaviour that is on bleed, which probably means that this part of the code is behaving correctly, but that `ToFacing` has an unexpected value &rarr; likely related to the classic facing fudge.

Reply:
I think the problem was that `var ret` would make the `OnComplete` pass an outdated `mobile.Facing` to the next MovePart, since `var ret` is created before `UpdateCenterLocation` runs.

Updated, hopefully this is the correct fix now.